src/character.cpp

```diff
@@ -1847,6 +1849,80 @@ nc_color Character::symbol_color() const
     return basic;
 }
 
+bool Character::is_dangerous_field( const field &fd ) const
+{
+    if( fd.fieldCount() == 0 || has_trait( debug_nodmg ) ) {
+        return false;
+    }
+
+    for( auto &fld : fd ) {
+        if( is_dangerous_field( fld.second ) ) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool Character::is_dangerous_field( const field_entry &entry ) const
+{
+    const field_id fid = entry.getFieldType();
+    switch( fid ) {
+        // @todo Lower density fields are less dangerous
+        case fd_smoke:
```

Comment:
I left it separate for the `@todo`. Field id itself doesn't have any density data and as a result is needlessly strict about how dangerous the fields are.